sound/soc/bcm/allo-piano-dac-plus.c

```diff
@@ -111,13 +113,35 @@ static int __snd_allo_piano_dsp_program(struct snd_soc_pcm_runtime *rtd,
 		return 0;
 
 	if (mode == 0) { /* 2.0 */
+		snd_soc_write(rtd->codec_dais[0]->codec,
+				PCM512x_MUTE, 0x00);
 		snd_soc_write(rtd->codec_dais[1]->codec,
 				PCM512x_MUTE, 0x11);
 		glb_ptr->set_rate = rate;
 		glb_ptr->set_mode = mode;
 		glb_ptr->set_lowpass = lowpass;
 		return 1;
+	} else if (mode == 3) { /* dual Stereo */
+		snd_soc_write(rtd->codec_dais[0]->codec,
+				PCM512x_MUTE, 0x00);
+		snd_soc_write(rtd->codec_dais[1]->codec,
+				PCM512x_MUTE, 0x00);
+		glb_ptr->set_rate = rate;
+		glb_ptr->set_mode = mode;
+		glb_ptr->set_lowpass = lowpass;
+		return 1;
+	} else if (mode == 4) { /* dual Mono */
+		snd_soc_write(rtd->codec_dais[0]->codec,
+				PCM512x_MUTE, 0x01);
+		snd_soc_write(rtd->codec_dais[1]->codec,
+				PCM512x_MUTE, 0x10);
+		glb_ptr->set_rate = rate;
+		glb_ptr->set_mode = mode;
+		glb_ptr->set_lowpass = lowpass;
+		return 1;
 	} else {
```

Comment:
Since we're doing a full review...

All success paths include:
```
	glb_ptr->set_rate = rate;
	glb_ptr->set_mode = mode;
	glb_ptr->set_lowpass = lowpass;
```
The code would be shorter and clearer if these were factored out, either with a "success" label or initialising them assuming success, then resetting at least "rate" on error to force a reinitialisation next time.